### PR TITLE
feat: Added renovate configuration

### PR DESCRIPTION
**Pull request by kvanzuijlen**
## Description
This PR adds the configuration file so that we can start using [Renovate](https://docs.renovatebot.com/) to keep dependencies up-to-date. Once the PR is merged, the [Renovate GitHub app](https://github.com/apps/renovate) should be enabled.

## Motivation and Context
We can use Renovate to keep all dependencies in oauth2-proxy up-to-date. Renovate is also able to include changelogs for some dependencies in the PR description.

## How Has This Been Tested?
Running renovate locally using `LOG_LEVEL=debug renovate --platform=local`

## Checklist:

- [ ] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.
- [ ] I have written tests for my code changes.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -5,29 +5,29 @@ This website is built using [Docusaurus 2](https://v2.docusaurus.io/), a modern
 ## Installation
 
 ```console
-yarn install
+npm install
```
Comment (kvanzuijlen) [inline docs/README.md:8]: Saw this says yarn, although there's only a package-lock.json committed.

**Comment by tuunit** [comment]
Why wouldn't we just use the official GitHub Dependabot instead?

**Comment by kvanzuijlen** [comment]
> Why wouldn't we just use the official GitHub Dependabot instead? What's the benefit of renovate?

Its far less comfigurable. We wouldn't be able to keep the alpine base image up to date for example.

**Review by tuunit — APPROVED** [review]
LGTM

**Comment by kvanzuijlen** [comment]
@JoelSpeed can you let me know when the following has happened?
> This PR adds the configuration file so that we can start using [Renovate](https://docs.renovatebot.com/) to keep dependencies up-to-date. Once the PR is merged, the [Renovate GitHub app](https://github.com/apps/renovate) should be enabled.

**Comment by JoelSpeed** [comment]
Already done, and already merged a couple of go.mod bumps

**Pull request merged**